src/Cake.Core/IO/NuGet/IAssemblyFilePathFrameworkNameParser.cs

```diff
@@ -0,0 +1,17 @@
+using System.Runtime.Versioning;
+
+namespace Cake.Core.IO.NuGet
+{
+    /// <summary>
+    /// Represents an object that parses the segments of a filepath for a .Net framework name
+    /// </summary>
+    public interface IAssemblyFilePathFrameworkNameParser
+    {
+        /// <summary>
+        /// Parses the framework name from assembly file path.
+        /// </summary>
+        /// <param name="path">The assembly file path.</param>
+        /// <returns>the parsed framework name -- null when path contains no folders</returns>
```

Comment:
`<returns>the parsed framework name, or <c>null</c> when path contains no folders.</returns>`